provisioner/clusterpy.go

```diff
@@ -1199,3 +1225,38 @@ func int32Value(v *int32) int32 {
 	}
 	return 0
 }
+
+func groupNodePools(logger *log.Entry, cluster *api.Cluster, updater updatestrategy.UpdateStrategy) []nodePoolGroup {
+	var masters, workers []*api.NodePool
+	for _, nodePool := range cluster.NodePools {
+		if strings.HasPrefix(nodePool.Profile, "master") {
+			masters = append(masters, nodePool)
+			continue
+		}
+
+		workers = append(workers, nodePool)
+	}
+
+	return []nodePoolGroup{
+		{
+			NodePools: masters,
+			Updater:   updater,
+			ReadyFn: func() error {
+				return waitForAPIServer(logger, cluster.APIServerURL, 15*time.Minute)
+			},
+		},
+		{
+			NodePools: workers,
+			Updater:   updater,
+			ReadyFn: func() error {
+				return nil
+			},
+		},
+	}
+}
+
+type nodePoolGroup struct {
+	NodePools []*api.NodePool
+	Updater   updatestrategy.UpdateStrategy
```

Comment:
We only have one right now (I just didn't have time to get rid of it), is it really needed?

Reply:
Good point, dropped it from here. Initially I was implementing a different (no-op) updater but this is now changed so no need to carry it per node pool group.